Applications/Utils/SWMMConverter/SWMMConverter.cpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/FileIO/SWMM/SWMMInterface.h"
+#include "ThirdParty/SWMMInterface/swmm5_iface.h"
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "BaseLib/StringTools.h"
+
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/IO/XmlIO/Boost/BoostXmlGmlInterface.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Properties.h"
+#include "MeshLib/IO/VtkIO/VtuInterface.h"
+
+#include "Applications/FileIO/CsvInterface.h"
+
+int main(int argc, char *argv[])
+{
+    ApplicationsLib::LogogSetup setup;
+
+    TCLAP::CmdLine cmd
+        ("Read files for the Storm Water Management Model (SWMM) and converts them to OGS.", ' ', "0.1");
+    TCLAP::ValueArg<std::string> mesh_output_arg
+        ("m","mesh", "mesh output file (*.vtu)", false, "", "mesh output file");
+    cmd.add(mesh_output_arg);
+    TCLAP::ValueArg<std::string> geo_output_arg
+        ("g","geo", "geometry output file (*.gml)", false, "", "geometry output file");
+    cmd.add(geo_output_arg);
+    TCLAP::ValueArg<std::string> swmm_input_arg
+        ("i","input", "SWMM input file (*.inp)", true, "", "input file");
+    cmd.add(swmm_input_arg);
+    cmd.parse( argc, argv );
+
+    if (!(geo_output_arg.isSet() || mesh_output_arg.isSet()))
+    {
+        ERR ("No output format given. Please specify OGS geometry or mesh output file.");
+        return -1;
+    }
+
+    if (geo_output_arg.isSet())
+    {
+        GeoLib::GEOObjects geo_objects;
+        if (!SwmmInterface::SwmmInputToGeometry(swmm_input_arg.getValue(), geo_objects, true))
+            return -1;
+
+        GeoLib::IO::BoostXmlGmlInterface xml(geo_objects);
+        xml.setNameForExport(BaseLib::extractBaseNameWithoutExtension(swmm_input_arg.getValue()));
+        xml.writeToFile(geo_output_arg.getValue());
+        return 0;
+    }
+
+    SwmmInterface* swmm = nullptr;
+    MeshLib::Mesh* mesh = nullptr;
+    if (mesh_output_arg.isSet())
+    {
+        swmm = SwmmInterface::create(swmm_input_arg.getValue());
+        std::size_t a = swmm->getNObjects(SwmmObject::SUBCATCHMENT);
+        if (swmm == nullptr)
+            return -1;
+
+        mesh = swmm->getMesh();
+        MeshLib::IO::VtuInterface vtkIO(mesh, 0, false);
+        vtkIO.writeToFile(mesh_output_arg.getValue());
+    }
+
+    std::cout << "Simulation time steps: " << swmm->getNTimeSteps() << std::endl;
+
+/*
```

Comment:
This shows basically how to use the interface for writing csv and meshes with scalar data. I could uncomment it but given the examples I have that'd mean that the utility would write 1000s of files (> 1 gb) to your hdd without warning.

I could also remove the code but then there's no example how to use the interface.